README.md

```diff
@@ -356,7 +361,98 @@ $ ./bin/ipc-agent leave-subnet --subnet=/root/t01002
+>>> Default wallet: t1ivy6mo2ofxw4fdmft22nel66w63fb7cuyslm4cy
+>>> Subnet subnet validator info:
+t1ivy6mo2ofxw4fdmft22nel66w63fb7cuyslm4cy@/ip4/172.17.0.4/udp/1348/quic/p2p/12D3KooWEJXcSPw6Yv4jDk52xvp2rdeG3J6jCPX9AgBJE2mRCVoR
+t1ivy6mo2ofxw4fdmft22nel66w63fb7cuyslm4cy@/ip4/127.0.0.1/udp/1348/quic/p2p/12D3KooWEJXcSPw6Yv4jDk52xvp2rdeG3J6jCPX9AgBJE2mRCVoR
+t1ivy6mo2ofxw4fdmft22nel66w63fb7cuyslm4cy@/ip4/172.17.0.4/tcp/1347/p2p/12D3KooWEJXcSPw6Yv4jDk52xvp2rdeG3J6jCPX9AgBJE2mRCVoR
+t1ivy6mo2ofxw4fdmft22nel66w63fb7cuyslm4cy@/ip4/127.0.0.1/tcp/1347/p2p/12D3KooWEJXcSPw6Yv4jDk52xvp2rdeG3J6jCPX9AgBJE2mRCVoR
+>>> API listening in host port 1240
+>>> Validator listening in host port 1359
+```
+
+#### Configuring the agent
+To configure the agent for its use with all the validators, we need to connect to the RPC API of one of the validators, and import all of the wallets of the validators in that node, so the agent is able through the same API to act on behalf of any validator. More information about importing [keys can be found in this section](#Importing-a-wallet-to-a-subnet-node).
+
+Here's an example of the configuration connecting to the RPC of the first validator, and configuring all the wallets for the validators in the subnet.
+```toml
+[[subnets]]
+id = "/root/t01002"
+gateway_addr = "t064"
+network_name = "test"
+jsonrpc_api_http = "http://127.0.0.1:1240/rpc/v1"
+auth_token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJBbGxvdyI6WyJyZWFkIiwid3JpdGUiLCJzaWduIiwiYWRtaW4iXX0.JTiumQwFIutkTb0gUC5JWTATs-lUvDaopEDE0ewgzLk"
+accounts = ["t1ivy6mo2ofxw4fdmft22nel66w63fb7cuyslm4cy", "t1cp4q4lqsdhob23ysywffg2tvbmar5cshia4rweq", "t1nv5jrdxk4ljzndaecfjgmu35k6iz54pkufktvua", "t1pr3qzqieikp33pfdaygwkwt5v35a5ljsxsu64xq", "t1pmxk3nhg2u2ge7ioilrk7c5rmdmmbgpazipvzyy"]
+```
+Remember to run `./bin/ipc-agent reload-config` for your agent to pick up the latest changes for the config.
+
+#### Joining the subnet
+All the infrastructure for the subnet is now deployed, and we can join our validators to the subnet. For this, we need to send a `join` command from each of our validators from their validator wallet addresses providing the validators multiaddress. We can get the validator multiaddress from the output of the script we ran to deploy the infrastructure (or by running `eudico mir validator config validator-addr`). These scripts expose through docker the port for the TPC port of the libp2p host of the validator (if there is interest to use the UDP-based multiaddress drop an issue and we can update them accordingly). Hence, we need to choose from the available multiaddresses one that exposes a TCP connection. Finally, our validators are configured so their docker containers interact with each other through ports in the host machine, so instead of exposing an IP address for the multiaddress, the following DNS-based multiaddress needs to be used for all validators: `/dns/host.docker.internal/`.
```

Comment:
I'll take your word for it but, in the previous tests, things always worked for me with IP address. There's probably a way to make either work, but, as it stands, I don't think the instructions do, mainly because eg

```
>>> Subnet /root/t01015 daemon running in container: 975db017f41c41ca423fa04205add5f89bc28a8c68c4eadac06e044d0e8d3c5f (friendly name: ipc_root_t01015_1240)
>>> Token to /root/t01015 daemon: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJBbGxvdyI6WyJyZWFkIiwid3JpdGUiLCJzaWduIiwiYWRtaW4iXX0.GjgfYovLNSkVZmO7AoVu-YJeqv1A_s0BvZSsR54xtEg
>>> Default wallet: t1aa4er7k6c7jjcv6vmdz7wtm76qhj3x3qp4ltj2q
>>> Subnet subnet validator info:
t1aa4er7k6c7jjcv6vmdz7wtm76qhj3x3qp4ltj2q@/ip4/172.17.0.2/tcp/1347/p2p/12D3KooWMwhd7SREGYj3ZSSWRdtdC4Y51dxdWBF3UZgQXzWb9PQE
t1aa4er7k6c7jjcv6vmdz7wtm76qhj3x3qp4ltj2q@/ip4/127.0.0.1/tcp/1347/p2p/12D3KooWMwhd7SREGYj3ZSSWRdtdC4Y51dxdWBF3UZgQXzWb9PQE
t1aa4er7k6c7jjcv6vmdz7wtm76qhj3x3qp4ltj2q@/ip4/172.17.0.2/udp/1348/quic/p2p/12D3KooWMwhd7SREGYj3ZSSWRdtdC4Y51dxdWBF3UZgQXzWb9PQE
t1aa4er7k6c7jjcv6vmdz7wtm76qhj3x3qp4ltj2q@/ip4/127.0.0.1/udp/1348/quic/p2p/12D3KooWMwhd7SREGYj3ZSSWRdtdC4Y51dxdWBF3UZgQXzWb9PQE
>>> API listening in host port 1240
>>> Validator listening in host port 1359
```

The multiaddr uses the same ports for every node (1347, 1348) and not the docker exposed ports (1240, 1359). So if you take the addresses output by eudico and just replace the IP address with `/dns/host.docker.internal/`, you'll get, for every node:

` --validator-net-addr="/dns/host.docker.internal/tcp/1347/p2p/<peer id>"`

Which evidently can't work. So we need to either tell people to use the right ports or the right address.

Reply:
Related potential bad idea... Should we change the deployment script again to filter the validator-addr output and replace it to yield the output people will actually need? Something like...

```
docker exec -it $img eudico mir validator config validator-addr | grep '127.0.0.1/tcp' | sed 's/^.*\/127.0.0.1/\/dns\/host.docker.internal/' | sed -E "s/\/[0-9]+\//\/$PORT\//"
```

yielding example output

```
/dns/host.docker.internal/tcp/1240/p2p/12D3KooWMwhd7SREGYj3ZSSWRdtdC4Y51dxdWBF3UZgQXzWb9PQE
```

instead of the current 4 lines

🤷

Reply:
> 'll take your word for it but, in the previous tests, things always worked for me with IP address.
This is because you were running a subnet with a single validator that doesn't need to dial itself, so as long as you provide a valid multiaddr for its peerID it will work, but if you run all validators on a docker container and try to dial other containers it'll fail. 


I agree with all of your points, and this one is also a hard one. I wanted to strike the right balance between completely abstracting everything from the user and giving enough information for someone not using Docker and trying to make a production deployment to know what is happening under the hood. You renaming proposals is probably the best approach (and the one I like the most). That way users can just copy-paste the address and that's it. Do you mind adding an issue so we tackle in the next iteration? (that way I can merge this one and cut a release before we start the next milestone). Thanks :pray:

Reply:
I can, but I'd still really like to run the multinode deployment to successful completion. I'm in back-to-back meetings but will try to do it during the day. Or you can merge and I'll provide feedback afterwards. What's your ETA for the release?

Reply:
I can wait, no rush. Feel free to ping me if you want to sync and unblock this.

Reply:
PR here: https://github.com/consensus-shipyard/lotus/pull/145

Not strongly attached to it, feel free to merge or close at your convenience.